### ISSUE TITLE
Unnecessary dependencies on vue compiler

### ISSUE THREAD

**Issue by lunny** [lafriks/vue-bar-graph#18]
ref: https://github.com/go-gitea/gitea/pull/31705#issuecomment-2251807259

**Comment by lafriks** [comment]
What exactly needs to be done about that? 🤔

**Comment by silverwind** [comment]
I'm not sure but I think the module entry file should maybe just export the SFC alone, if that is possible.

```js
import VueBarGraph from './components/VueBarGraph.vue';
export {VueBarGraph};
```

Check how other vue component libraries export their components.

**Comment by silverwind** [comment]
Also I guess `vue` should be in [`peerDependencies`](https://docs.npmjs.com/cli/v10/configuring-npm/package-json#peerdependencies).

**Comment by aruddick-copia** [comment]
> Also I guess `vue` should be in [`peerDependencies`](https://docs.npmjs.com/cli/v10/configuring-npm/package-json#peerdependencies).

Yes!  I just tracked down a bug in my install because my bundler tried to pull in two copies of `vue` that conflicted.
Using `external: 'vue'` in `vite.config.mjs` helps, but only if you're using vite.
